src/core/api/search.js

```diff
@@ -37,6 +41,22 @@ export function search({ api, auth = false, filters = {} }: SearchParams) {
     userAgentInfo: api.userAgentInfo,
   });
 
+  // The API says: 'The "sort" parameter "random" can only be specified when
+  // the "featured" or "recommended" parameter is also present, and the "q"
+  // parameter absent.'
+  // Let's make sure we don't send invalid filters.
+  if (newFilters.sort && newFilters.sort === SEARCH_SORT_RANDOM) {
+    if (
+      !(newFilters.recommended || newFilters.featured) ||
+      ((newFilters.recommended || newFilters.featured) && newFilters.q)
+    ) {
+      delete newFilters.sort;
+
+      log.warn(oneLine`search api filter "sort=random" has been removed before
```

Comment:
Would it be worth moving this `log` statement above the `delete` and including `newFilters` in the statement so we can see when a case arises that requires this, and what the data looked like?

Reply:
so we don't really want to log objects because it has performance implications, especially here since we can have user input (and potentially very large input).